### PR TITLE
Fix time range update for variables

### PR DESCRIPTION
**Pull request by Loori-R**
This PR resolves issue #47 by ensuring variables correctly update to match the user-selected time range.

### CODE REVIEW

File: src/services/TimeSrv.ts [hunk 1]
```diff
@@ -378,7 +378,9 @@ export class TimeSrv {
   };
 
   timeRange(): TimeRange {
-    return getTimeRange(this.time, this.timeModel);
+    // @ts-ignore default time
+    const time = getTemplateSrv()?.timeRange?.raw || getDefaultTimeRange().raw;
+    return getTimeRange(time, this.timeModel)
```
Comment (dmitryk-dk) [inline src/services/TimeSrv.ts:383]: Is it possible to reassign `this.time` to avoid using `// @ts-ignore default time`?

Comment (Loori-R) [reply src/services/TimeSrv.ts]: updated

**Review by dmitryk-dk — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## tip
 
+* BUGFIX: respect time filter for variables update. See [this issue](https://github.com/VictoriaMetrics/grafana-datasource/issues/47)
```
Comment (hagen1778) [inline CHANGELOG.md:5]:
```suggestion
* BUGFIX: respect the time filter change on updating dashboard variables. See [this issue](https://github.com/VictoriaMetrics/grafana-datasource/issues/47)
```

**Review by hagen1778 — APPROVED** [review]
LGTM!

**Review by hagen1778 — APPROVED** [review]

**Pull request merged**